tests/Unit/IO/Test_VolumeData.py

```diff
@@ -15,25 +15,88 @@ class TestIOH5VolumeData(unittest.TestCase):
+        vol_file = h5_file.get_vol(path="/element_data")
+        # Set TensorComponent and ExtentsAndTensorVolumeData to
+        # be written
+        extents_and_tensor_vol_obs1 = [
+            ds.ExtentsAndTensorVolumeData([2, 2, 2], [
+                ds.TensorComponent(grid_names[0] + "/S",
+                                   self.tensor_components_and_coords[0]),
+                ds.TensorComponent(grid_names[0] + "/x-coord",
+                                   self.tensor_components_and_coords[1]),
+                ds.TensorComponent(grid_names[0] + "/y-coord",
+                                   self.tensor_components_and_coords[2]),
+                ds.TensorComponent(grid_names[0] + "/z-coord",
+                                   self.tensor_components_and_coords[3]),
+                ds.TensorComponent(grid_names[0] + "/T_x",
+                                   self.tensor_components_and_coords[4]),
+                ds.TensorComponent(grid_names[0] + "/T_y",
+                                   self.tensor_components_and_coords[5]),
+                ds.TensorComponent(grid_names[0] + "/T_z",
+                                   self.tensor_components_and_coords[6])
+            ]) for observation_value in observation_values
+        ]
+
+        extents_and_tensor_vol_obs2 = [
+            ds.ExtentsAndTensorVolumeData([2, 2, 2], [
+                ds.TensorComponent(grid_names[1] + "/S",
+                                   self.tensor_components_and_coords[0]),
+                ds.TensorComponent(grid_names[1] + "/x-coord",
+                                   self.tensor_components_and_coords[2]),
+                ds.TensorComponent(grid_names[1] + "/y-coord",
+                                   self.tensor_components_and_coords[3]),
+                ds.TensorComponent(grid_names[1] + "/z-coord",
+                                   self.tensor_components_and_coords[1]),
+                ds.TensorComponent(grid_names[1] + "/T_x",
+                                   self.tensor_components_and_coords[6]),
+                ds.TensorComponent(grid_names[1] + "/T_y",
+                                   self.tensor_components_and_coords[4]),
+                ds.TensorComponent(grid_names[1] + "/T_z",
+                                   self.tensor_components_and_coords[5])
+            ]) for observation_value in observation_values
+        ]
+        # Write extents and tensor volume data to volfile
+        vol_file.write_volume_data(
+            observation_ids[0], observation_values[0],
+            [extents_and_tensor_vol_obs1[0], extents_and_tensor_vol_obs2[0]])
+        vol_file.write_volume_data(
+            observation_ids[1], observation_values[1],
+            [extents_and_tensor_vol_obs1[1], extents_and_tensor_vol_obs2[1]])
+        self.h5_file = h5_file
 
     def tearDown(self):
-        if os.path.isfile(self.file_name_w):
-            os.remove(self.file_name_w)
+        if os.path.isfile(self.file_name):
```

Comment:
Add `self.h5_file.close()` in the line above. Then you don't need to do that in each test.